### ISSUE TITLE
Proposal: Add a method to directly create an `arrow::RecordBatch`

### ISSUE THREAD

**Issue by alamb** [chmp/serde_arrow#132]
Again, thank you for this library.

`arrow` makes heavy use of [`RecordBatch`](https://docs.rs/arrow/latest/arrow/array/struct.RecordBatch.html) (while `arrow2` removed the equivalent concept and deals in `Vec`s of  `Arrays`). 

It is possible to make `RecordBatch` from the APIs in this crate, as I demonstrate in https://github.com/chmp/serde_arrow/pull/131, I think it would a nicer API if this crate offered a convenience method for doing so directly

### Here is what you have to do today (copied from https://github.com/chmp/serde_arrow/pull/131)

```rust
// Determine Arrow schema
let fields =
  SerdeArrowSchema::from_type::<Record>(TracingOptions::default())?
  .to_arrow_fields()

// Convert to Arrow arrays
let arrays = serde_arrow::to_arrow(&fields, &records)?;

// Form a RecordBatch
let schema = Schema::new(&fields);
let batch = RecordBatch::try_new(schema.into(), arrays)?;
```

### Proposal
I propose adding a `to_batch` method that coverts directly to `arrow` RecordBatch:

```rust
pub fn to_batch<T: Serialize + ?Sized>(fields: &[Field], items: &T) -> Result<RecordBatch> {
...
}
```

Example use:

```rust
// Determine Arrow schema
let fields =
  SerdeArrowSchema::from_type::<Record>(TracingOptions::default())?
  .to_arrow_fields()

// Convert to RecordBatch
let batch = serde_arrow::to_batch(&fields, &records)?;

```

**Comment by alamb** [comment]
If this is acceptable to you @chmp  I would be happy to create a PR with a proposal

**Comment by chmp** [comment]
Hey @alamb. 

Thanks for the offer, in principle I would be open to an PR, but only reluctantly.

To explain my reservations: due to [the constant API churn](https://github.com/apache/arrow-rs/issues/1120) I consider a larger exposure to the arrow API a huge risk. Initially `serde_arrow` was built around the `arrow` API, but after multiple updates breaking my code (within a six week timestamp), I took the decision to minimize the API used. That's the reason why all builders are implemented from scratch. That's the reason why I **removed** the record batch based API in `serde_arrow` some time ago. That's the reason why there is this whole [non-additive feature issue](https://github.com/chmp/serde_arrow/issues/52).

**Comment by alamb** [comment]
No worries -- your explantion makes sense. I think having an example (e.g. https://github.com/chmp/serde_arrow/pull/131) would provide 90% of the value. Maybe over time depending on how peple use the crate it would make sense to offer additional APIs, but I can see the value in keeping things simple

**Comment by chmp** [comment]
See also #137

**Comment by chmp** [comment]
Added a `to_record_batch` on main

**Comment by chmp** [comment]
Released as `serde_arrow=0.11.0`
